### PR TITLE
Replaces the changeling spacesuit with a passive ability

### LINKED ISSUE #74168: Organic space suit does not regulate oxygen needs as advertised.
Reporting client version: 514.1589

## Round ID:
[202258](https://scrubby.melonmesa.com/round/202258)

## Reproduction:
The description for the changeling's ability claims it regulates the oxygen needs, but this does not work as advertised. Oxygen is not taken care of and other gases can also pose an issue resulting in space changelings dying repeatedly as they scramble to find an airlock.

### PR DESCRIPTION
**Pull request by Jacquerel**
## About The Pull Request

Fixes #74168

I was going to make changes to the changeling spacesuit so that it works on Icebox but then I thought, why not _not_ do that.
This isn't a commonly picked adaption so why don't we make it a little better.

What's more spooky, hearing a knock on the window and seeing a fat suit outside? Or this?
image

Picking Void Adaption will now make you immune to low temperature and pressure (but not high temperature and pressure) and you will stop breathing. If you enter an area with low temperature or pressure then your chemical regeneration rate will decrease until you leave that area. Because it doesn't put a suit on you, it now also works during Lesser Form.

While testing this I noticed that we weren't calling `Grant` on passive changeling abilities for some reason, I replaced that with the already-written interface for making "an action which doesn't give you a button".

If people really _really_ miss the fat suit I guess I'll rework that instead, but I think I like this more.

## Why It's Good For The Game

Makes a niche-pick ability more useful and easier to use.
Meteor Changelings who land on Icebox now don't roll a dice to see if they get instantly knocked out by the atmosphere there.

## Changelog

:cl:
balance: The Changeling Space Suit has been replaced by a new ability which makes you passively spaceproof without replacing your clothing.
admin: Editing the atmos sensitivity variables on a basic mob during the game will now actually do something.
/:cl:

### CODE REVIEW

**Comment by Sealed101** [comment]
closes #74168 since you don't breathe

**Comment by CRITAWAKETS** [comment]
this should probably not turn on if you're in a hazardous area but have the protective gear for it like a space suit.

**Comment by Jacquerel** [comment]
> this should probably not turn on if you're in a hazardous area but have the protective gear for it like a space suit.

Already did for pressure, I will make sure it also does for temperature.

**Comment by SimplySmiley123** [comment]
Honestly seems amazing.
The fat suit is, genuinely pointless with fleshmend giving you more then enough time you'll ever need for space walking, plus stealing EVA isn't that hard anyway, nor is using ghetto EVA methods.
And outside of the skin suit normal crew can get without a ling present, doubt anyone would miss the fat suit on lings.

File: code/modules/antagonists/changeling/changeling.dm [hunk 1]
```diff
@@ -1049,8 +1048,8 @@
 /datum/outfit/changeling_space
 	name = "Changeling (Space)"
 
-	head = /obj/item/clothing/head/helmet/space/changeling
-	suit = /obj/item/clothing/suit/space/changeling
+	head = /obj/item/clothing/head/helmet/space
+	suit = /obj/item/clothing/suit/space
```
Comment (Fikou) [inline code/modules/antagonists/changeling/changeling.dm:1052]: that's pretty lame, why would an organic fleshy meteor spawn you with a space suit? should either be the chitin armor or just you being naked

Comment (Fikou) [reply code/modules/antagonists/changeling/changeling.dm:1052]: (or make the changeling spawn as a headslug but i think people would be upset)

File: code/modules/mob/living/basic/basic.dm [hunk 2]
```diff
@@ -260,3 +260,9 @@
 	else if(on_fire && !isnull(last_icon_state))
 		return last_icon_state
 	return null
+
+/mob/living/basic/get_body_temp_heat_damage_limit()
+	return maximum_survivable_temperature
+
+/mob/living/basic/get_body_temp_cold_damage_limit()
+	return minimum_survivable_temperature
```
Comment (Fikou) [inline code/modules/mob/living/basic/basic.dm:268]: not a fan, they are meant to be template vars for easy applying.

**Review by Fikou — CHANGES_REQUESTED** [review]
.

File: code/modules/antagonists/changeling/changeling.dm (hunk 1 above)
Comment (Jacquerel) [reply code/modules/antagonists/changeling/changeling.dm:1052]: Sure I can make you nude instead

File: code/modules/mob/living/basic/basic.dm (hunk 2 above)
Comment (Jacquerel) [reply code/modules/mob/living/basic/basic.dm:268]: I don't know how else I would do this without calling `GetComponent` and reading the value from the component, which cannot be changed from this value at runtime anyway except through VV.

Comment (Jacquerel) [reply code/modules/mob/living/basic/basic.dm:292]: we figured out a compromise on discord

**Pull request merged**